src/components/menu-item/menu-item.tsx

```diff
@@ -0,0 +1,469 @@
+          children: this.subMenuItems,
+          isSubMenuOpen: this.open && this.hasSubMenu
+        });
+        break;
+    }
+  };
+
+  private selectMenuItem(event: MouseEvent | KeyboardEvent): void {
+    if (event.target !== this.dropDownActionEl) {
+      this.calciteMenuItemSelect.emit();
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  //--------------------------------------------------------------------------
+
+  renderIconElStart(): VNode {
+    return (
+      <calcite-icon
+        class={{
+          [CSS.icon]: true,
+          [CSS.iconStart]: true
+        }}
+        flipRtl={this.iconFlipRtl === "start" || this.iconFlipRtl === "both"}
+        icon={this.iconStart}
+        scale="s"
+      />
+    );
+  }
+
+  renderIconElEnd(): VNode {
+    return (
+      <calcite-icon
+        class={{
+          [CSS.icon]: true,
+          [CSS.iconEnd]: true
+        }}
+        flipRtl={this.iconFlipRtl === "end" || this.iconFlipRtl === "both"}
+        icon={this.iconEnd}
+        scale="s"
+      />
+    );
+  }
+
+  renderBreadcrumbIcon(dir: Direction): VNode {
+    return (
+      <calcite-icon
+        class={{
+          [CSS.icon]: true,
+          [CSS.iconBreadcrumb]: true
+        }}
+        icon={dir === "rtl" ? "chevron-left" : "chevron-right"}
+        scale="s"
+      />
+    );
+  }
+
+  renderDropdownIcon(dir: Direction): VNode {
+    const dirChevron = dir === "rtl" ? "chevron-left" : "chevron-right";
+    return (
+      <calcite-icon
+        class={{
+          [CSS.icon]: true,
+          [CSS.iconDropdown]: true
+        }}
+        icon={
+          this.topLevelMenuLayout === "vertical" || this.isTopLevelItem
+            ? this.open
+              ? "chevron-up"
+              : "chevron-down"
+            : dirChevron
+        }
+        id="render-dropdown-icon"
+        scale="s"
+      />
+    );
+  }
+
+  renderDropdownAction(dir: Direction): VNode {
+    const dirChevron = dir === "rtl" ? "chevron-left" : "chevron-right";
+    return (
+      <calcite-action
+        class={CSS.dropdownAction}
+        icon={
+          this.topLevelMenuLayout === "vertical" || this.isTopLevelItem
+            ? this.open
+              ? "chevron-up"
+              : "chevron-down"
+            : dirChevron
+        }
+        onClick={this.clickHandler}
+        onKeyDown={this.keyDownHandler}
+        text="open-dropdown"
+        // eslint-disable-next-line react/jsx-sort-props
+        ref={(el) => (this.dropDownActionEl = el)}
+      />
+    );
+  }
+
+  renderSubMenuItems(dir: Direction): VNode {
+    return (
+      <calcite-menu
+        class={{
+          [CSS.dropdownMenuItems]: true,
+          [CSS.open]: this.open,
+          [CSS.nested]: !this.isTopLevelItem,
+          [CSS.isRtl]: dir === "rtl",
+          [CSS.dropdownVertical]: this.topLevelMenuLayout === "vertical"
+        }}
+        label="Submenu"
```

Comment:
This label needs a nls string right?

Reply:
yup.

Reply:
This label string still needs to be translated.